src/matching_market.sol

```diff
@@ -45,17 +47,28 @@ contract MatchingMarket is MatchingEvents, ExpiringMarket, DSNote {
     mapping(address => uint) public _dust;                      //minimum sell amount for a token to avoid dust offers
     mapping(uint => uint) public _near;         //next unsorted offer id
     uint _head;                                 //first unsorted offer id
-    uint public dustId;                         // id of the latest offer marked as dust
 
+    // dust management
+    address DAI;
+    uint256 daiDustLimit;
+    UniswapSimplePriceOracle uniswapSimplePriceOracle;
 
-    constructor(uint64 close_time) ExpiringMarket(close_time) public {
+    constructor(address _DAI, uint256 _daiDustLimit, UniswapSimplePriceOracle _uniswapSimplePriceOracle) public {
+        DAI = _DAI;
+        daiDustLimit = _daiDustLimit;
+        uniswapSimplePriceOracle = _uniswapSimplePriceOracle;
+
+        _setMinSell(ERC20(DAI), daiDustLimit);
     }
 
     // After close, anyone can cancel an offer
+    // If dust, anyone can cancel an offer
+    // If owner, can cancel an offer
     modifier can_cancel(uint id) {
         require(isActive(id), "Offer was deleted or taken, or never existed.");
+
         require(
-            isClosed() || msg.sender == getOwner(id) || id == dustId,
+            msg.sender == getOwner(id) || offers[id].pay_amt < _dust[address(offers[id].pay_gem)],
```

Comment:
should there be a mirror condition for a `buy_amt`?

Reply:
I suppose so, order can be dusty in two ways...